libretro-common/include/libretro.h

```diff
@@ -976,6 +977,85 @@ struct retro_hw_render_context_negotiation_interface
                                             * so it will be used after SET_HW_RENDER, but before the context_reset callback.
                                             */
 
+/* Seek flags from stdio.h */
+enum retro_seek_flag
+{
+  RETRO_SEEK_SET = 0, /* Beginning of file */
+  RETRO_SEEK_CUR = 1, /* Current position of the file pointer */
+  RETRO_SEEK_END = 2, /* End of file - implementation not required */
+};
+
+/* Stat fields */
+struct retro_stat_struct
+{
+  uint32_t deviceId;         /* ID of device containing file */
+  uint64_t size;             /* Total size, in bytes */
+  timespec accessTime;       /* Time of last access */
+  timespec modificationTime; /* Time of last modification */
+  timespec statusTime;       /* Time of last status change */
+  bool isDirectory;          /* The stat url is a directory */
+  bool isSymLink;            /* The stat url is a symbolic link */
+  bool isHidden;             /* The file is hidden */
+};
+
+/* Opaque file handle */
+typedef struct retro_file retro_file;
+
+/* File operations */
+typedef retro_file* (RETRO_CALLCONV *retro_open_file_t)(const char *url, int flags);
```

Comment:
- url is a strange name, I'd prefer path or something.
- You have not defined any flags.
- Those names feel way too long.
- Do we need both this and open_for_write?